src/utils/lzw.rs

```diff
@@ -268,8 +270,8 @@ impl EncodingDict {
     }
 }
 
-pub fn encode<R, W>(mut r: R, mut w: W, min_code_size: u8) -> old_io::IoResult<()>
-where R: Reader, W: BitWriter {
+pub fn encode<R, W>(mut r: R, mut w: W, min_code_size: u8) -> io::Result<()>
```

Comment:
This function has to be rewritten. `while let Ok(c) = ...` will be an infinite loop with `std::io`;